src/core/search/search.cc

```diff
@@ -409,8 +409,8 @@ std::vector<TextIndex*> FieldIndices::GetAllTextIndices() const {
   return out;
 }
 
-vector<DocId> FieldIndices::GetAllDocs() const {
-  return std::vector<DocId>{all_ids_.begin(), all_ids_.end()};
+const vector<DocId>& FieldIndices::GetAllDocs() const {
+  return all_ids_;
```

Comment:
I decided to remove the resource from the all-id vector. Because for 1M entries it'll take up only 4mb... which is small compared to what 1M entries probably take up.

I did this because I don't wanna complicate the search code. It uses `const vector<DocId>*` as a transparent type for accessing elements, but a vector with a polymorphic allocator can't easily be turned into such a vector... So I need to use absl::Span for everything which makes it more verbose